addons/crm/models/crm_lead.py

```diff
@@ -1206,6 +1335,354 @@ def get_import_templates(self):
+        so we can get everything we need for all leads to recompute in a minimum number of queries.
+        This batch mode is currently called when the computation is triggered by
+        crm.lead._cron_update_automated_probabilities().
+        :param team_id: (int) team_id to search on
+        :param batch_mode: (bool) batch mode
+        :return: dict of list of tuple of field - value by lead ({lead_id: [(field1: value1), (field2: value2), ...], ...})
+        """
+        leads_values_dict = {}
+        fields = ["stage_id"] + self._get_pls_safe_fields()
+        if batch_mode:
+            ids = tuple(self.ids) if len(self) > 1 else "(%s)" % self.id
+
+            #   Prepare team condition and params for both next queries
+            team_condition = 'and l.team_id = %s' if team_id else 'and l.team_id is null'
+            query_params = ([int(team_id)] if team_id else []) + [ids]
+
+            # get all info on leads
+            #   Prepare fields injection
+            str_fields = ", ".join(["{}"] * len(fields))
+            args = [sql.Identifier(field) for field in fields]
+            #   Build sql query in safe mode
+            query = """SELECT id, %s
+                        FROM crm_lead l
+                        WHERE probability > 0 AND probability < 100 %s AND active = True AND id in %%s"""
+            query = sql.SQL(query % (str_fields, team_condition)).format(*args)
+
+            self._cr.execute(query, query_params)
+            lead_results = self._cr.dictfetchall()
+
+            query = """SELECT l.id as lead_id, t.id as tag_id
+                        FROM crm_lead l
+                        LEFT JOIN crm_lead_tag_rel rel ON l.id = rel.lead_id
+                        LEFT JOIN crm_lead_tag t ON rel.tag_id = t.id
+                        WHERE l.probability > 0 AND l.probability < 100 %s AND l.active = True AND l.id in %%s"""
+            self._cr.execute(query % team_condition, query_params)
+            tag_results = self._cr.dictfetchall()
+
+            # get all (variable, value) couple for all in self
+            for lead in lead_results:
+                lead_values = []
+                for field in fields:
+                    value = lead[field]
+                    if field in ['is_phone_valid', 'is_email_valid'] or value:
+                        lead_values.append((field, value))
+                leads_values_dict[lead['id']] = lead_values
+
+            for tag in tag_results:
+                if tag['tag_id']:
+                    leads_values_dict[tag['lead_id']].append(('tag_id', tag['tag_id']))
+            return leads_values_dict
+        else:
+            for lead in self:
+                lead_values = [('stage_id', lead.stage_id.id)]
+                for field in fields:
+                    value = lead[field].id if isinstance(lead[field], models.BaseModel) else lead[field]
+                    if field in ['is_phone_valid', 'is_email_valid'] or value:
```

Comment:
why those 2 fields has a particular treatment ? If I want to add a custom boolean or interger field, will to be working ?